### PR TITLE
feat: port helpers for accesslist

### PR DESCRIPTION
**Pull request by mattsse**
prep for https://github.com/paradigmxyz/reth/pull/7554

adds a few helpers, could use a closer API review, perhaps some aren't even useful.

### CODE REVIEW

File: crates/eips/src/eip2930.rs [hunk 1]
```diff
@@ -101,6 +109,41 @@ impl AccessList {
...
 
+    /// Returns the position of the given address in the access list, if present.
+    pub fn index_of_address(&self, address: Address) -> Option<usize> {
+        self.iter().position(|item| item.address == address)
+    }
+
+    /// Checks if a specific storage slot within an account is present in the access list.
+    ///
+    /// Returns a tuple with flags for the presence of the account and the slot.
+    pub fn contains(&self, address: Address, slot: B256) -> (bool, bool) {
```
Comment (onbjerg) [inline crates/eips/src/eip2930.rs:120]: imo mirror other method so `contains_storage`? if not, for me the immediate expectation was that `contains` is what `contains_address` is

File: crates/eips/src/eip2930.rs [hunk 2]
```diff
@@ -101,6 +109,41 @@ impl AccessList {
...
+    }
+
+    /// Checks if the access list contains the specified address.
+    pub fn contains_address(&self, address: Address) -> bool {
+        self.iter().any(|item| item.address == address)
+    }
+
+    /// Checks if the storage keys at the given index within an account are present in the access
+    /// list.
+    pub fn contains_storage_key_at_index(&self, slot: B256, index: usize) -> bool {
```
Comment (onbjerg) [inline crates/eips/src/eip2930.rs:132]: what is this useful for vs `contains`?

File: crates/eips/src/eip2930.rs [hunk 3]
```diff
@@ -101,6 +109,41 @@ impl AccessList {
...
+        self.get(index).map_or(false, |entry| {
+            entry.storage_keys.iter().any(|storage_key| *storage_key == slot)
+        })
+    }
+
+    /// Adds an address to the access list and returns `true` if the operation results in a change,
+    /// indicating that the address was not previously present.
+    pub fn add_address(&mut self, address: Address) -> bool {
+        !self.contains_address(address) && {
+            self.0.push(AccessListItem { address, storage_keys: Vec::new() });
```
Comment (onbjerg) [inline crates/eips/src/eip2930.rs:142]: what is the use case for this? in my mind an access list is immutable unless it is being built by the node, which i assume would just work on a `Vec<AccessListItem>`

if it doesn't work on the `Vec`, should we add methods to add slots etc?

File: crates/eips/src/eip2930.rs (hunk 2 above)
Comment (mattsse) [reply crates/eips/src/eip2930.rs]: made private

**Review by onbjerg — APPROVED** [review]

**Pull request merged**